eeps/eep-0079.md

```diff
@@ -0,0 +1,802 @@
+otherwise, there will be a compilation error. There will be a
+compilation error if `Name` is neither the name of a local record nor
+an imported native record.
+
+If `is_record/2` is used in a function body, `Name` is allowed to be a
+variable.
+
+If `Name` refers to an imported native record, see the description of
+`is_record/3` for more details.
+
+Examples:
+
+```erlang
+-module(misc).
+-record #user() {a,b,c}.
+
+is_user(U) when is_record(U, user) ->
+    true;
+is_user(_U) ->
+    false.
+```
+
+<!-- -->
+
+```erlang
+-module(example).
+-import_record(misc, [user/0]).
+is_user(U) -> is_record(U, user).
+```
+
+#### `is_record/1`
+
+```erlang
+  -spec is_record(Term :: term()) -> boolean().
+```
+
+`is_record(Record)` returns `true` if `Record` is any native record value.
+
+### Native records in specs and in the language of types
+
+Native records that can be used as types using the following syntax:
+
+```erlang
+%% local or imported native-record
+#RecordName{TField :: TType, ... }
+%% remote native-record
+#Module:RecordName(TVar1, ..., TVarN)
+```
+
+If you export a native record, its type will be available for other
+modules to use.  Dialyzer will complain if you attempt to use an
+un-exported native record.
+
+Example:
+
+```erlang
+-module(misc).
+-export_record([user/0, pair/2]).
+-record #user() {
+    id = -1 :: integer(),
+    name :: binary(),
+    city :: binary()
+}.
+-record #pair(A, B) {
+    first :: A,
+    second :: B
+}.
+-type int_pair() :: #pair(integer(), integer()).
+-spec mk_user() -> #user().
+mk_user() ->
+    #user{id = 1, name = ~"Alice", city = ~"London"}.
+
+-spec mk_user_limited() -> #user().
+mk_user() ->
+    #user{id = 1, name = ~"Alice", city = ~"London"}.
+
+-spec mk_pair(A, B) -> #pair(A, B).
+mk_pair(A, B) ->
+    #pair{first = A, second = B}.
+```
+
+A new builtin type `record()` is introduced. It denotes the set of all
+possible native record values at runtime.
+
+### Documentation
+
+Native records can be documented just as functions/types/callbacks can be documented.
+If you export a record it will be visible and you have to add `-doc false.`
+for it to not be shown.
+
+If in a spec, type, callback, or native record refers to an undocumented
+local native record, the compiler will issue a warning.
+
+### Compatibility between OTP 28 and OTP 29
+
+When attempting to send a native record to an older node (OTP 28 or earlier),
+the sender should send a message to the logger process and close the connection.
+
+### Ordering and equality
+
+With addition of native records the runtime values of different types
+are now ordered as follows:
+
+```erlang
+number()
+< atom()
+< reference()
+< fun()
+< port()
+< pid()
+< tuple()
+< record()
+< map()
+< []
+< [_|_]
+< bitstring()
+```
+
+Native-record values are ordered by their fully qualified name, then
+by their visibility, then their keys, and finally by field values
+(field values are considered as maps). Equality is defined through
```

Comment:
What does "field values are considered as maps" mean?

Reply:
I will remove that part.